src/transaction.rs

```diff
@@ -766,7 +759,7 @@ mod tests {
         transport.add_response(json!(gas));
         transport.add_response(json!(gas_price));
         transport.add_response(json!(nonce));
-        transport.add_response(json!(format!("{}", chain_id))); // chain id
+        transport.add_response(json!(format!("0x{:x}", chain_id)));
```

Comment:
nit: you can use {:#x} to get the 0x prefix (cf. https://doc.rust-lang.org/nightly/core/fmt/trait.LowerHex.html)

Reply:
Woah... TIL!